helper/awsutil/generate_credentials.go

```diff
@@ -54,6 +59,20 @@ func (c *CredentialsConfig) GenerateCredentialChain() (*credentials.Credentials,
 			"static AWS client credentials haven't been properly configured (the access key or secret key were provided but not both)")
 	}
 
+	roleARN := os.Getenv("AWS_ROLE_ARN")
+	tokenPath := os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE")
+	if roleARN != "" && tokenPath != "" {
+		sessionName := os.Getenv("AWS_ROLE_SESSION_NAME")
+		// this session is only created to create the WebIdentityRoleProvider, as the env variables are already there
+		// this automatically assumes the role, but the provider needs to be added to the chain
+		sess, err := session.NewSession()
+		if err != nil {
+			return nil, errors.Wrap(err, "error creating a new session to create a WebIdentityRoleProvider")
+		}
+		//Add the environment credential provider
+		providers = append(providers, stscreds.NewWebIdentityRoleProvider(sts.New(sess), roleARN, sessionName, tokenPath))
```

Comment:
Would it be possible to get test coverage on this? One test where we set credentials for the WebIdentityRoleProvider and check everything was grabbed as expected, and one where we don't and we see what it defaults to? I realize we don't yet have tests on the `generate_credentials.go` file, but now would be a great time to start.